tests/manifests/test_deployments.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2025 New Vector Ltd
+#
+# SPDX-License-Identifier: AGPL-3.0-only
+
+import pytest
+
+from . import DeployableDetails, PropertyType, ValuesFilePath, values_files_to_test
+from .utils import iterate_deployables_workload_parts, template_id, template_to_deployable_details
+
+
+@pytest.mark.parametrize("values_file", values_files_to_test)
+@pytest.mark.asyncio_cooperative
+async def test_all_deployments_set_replicas(values, make_templates):
+    deployments = []
+    for template in await make_templates(values):
+        if template["kind"] == "Deployment":
+            deployments.append(template)
+
+    for deployment in deployments:
+        id = deployment["metadata"]["name"]
+        assert "replicas" in deployment["spec"], f"{id} does not specify replicas"
```

Comment:
```suggestion
async def test_all_deployments_set_replicas(templates):
    for template in templates:
        if template["kind"] in ["Deployment", "StatefulSet"]:
          assert "replicas" in template["spec"], f"{template_id(template)} does not specify replicas"
```

As this is a static test that isn't manipulating values and because `StatefulSets` should have this property too